etcdctl/command/exec_watch_command.go

```diff
@@ -51,21 +51,35 @@ func execWatchCommandFunc(c *cli.Context, ki client.KeysAPI) {
 		handleError(ExitBadArgs, errors.New("key and command to exec required"))
 	}
 
-	key := args[argslen-1]
-	cmdArgs := args[:argslen-1]
+	var (
+		key     string
+		cmdArgs []string
+	)
+
+	foundSep := false
+	for i := range args {
+		if args[i] == "--" && i != 0 {
```

Comment:
> i am totally in chaos.

This is a hack as before. This relies on the behavior of cli lib. It depends on how the cli lib parsing the flags. You have to read the etcdctl watch-exec doc to understand what is going on here.

> what is the meaning of args[0] == "--"? 
 args[0] is either your key or part of your command. The separator cannot be at position 0.